src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,1035 @@
+that property holds for a given integer.
+
+- None:  This is simply a placeholder to indicate that no further properties
+will be used.  Choosing anything other than none will add a new property
+and reveal parameters for it.
+- Prime:  Its absolute value is prime
+- Negative:  Less than zero
+- Even:  Divisible by two
+- Odd: Not even
+- Divisible_By:  Divisible by the specified divisor
+- Last_Digit_Is:  The final digit base 10 is the specified digit
+- Polygonal_Number:  Positive and that many dots can be arranged in a
+  polygonal arrangement with the specified number of sides.
+- Sum_Of_Two_Squares:  Nonnegative and equal to the sum of two squares
+- Abundant:  Its absolute value exceeds the sum of its proper divisors
+- Perfect:  Equal to the sum of its proper divisors
+- Deficient:  Its absolute value is less than the sum of its proper divisors
+- Semi_Prime:  Its absolute value is a
+  [semi-prime](https://en.wikipedia.org/wiki/Semiprime), that is, a product of
+  exactly two primes (possibly equal)
+
+##### Display:  Highlight style for cells with the property
+
+Using different display styles allows for visualizing two properties
+that are both true for the same value at once,
+without them overcoloring each other. (Note that later properties overcolor
+earlier ones that use the _same_ style.)
+
+- Fill_Cell:  Fill the complete cell
+- Box_In_Cell:  Fill only a smaller central box in the cell
+
+##### Color:  Highlight color for cells with the property
+         **/
+        for (let i = 0; i < MAXIMUM_ALLOWED_PROPERTIES; i++) {
+            const ithPropertyObject = {
+                property: i === 0 ? Property.Prime : Property.None,
+                visualization:
+                    i === 1
+                        ? PropertyVisualization.Box_In_Cell
+                        : PropertyVisualization.Fill_Cell,
+                color: DEFAULT_COLORS[i],
+                aux: 3n,
+            }
+            this.propertyObjects.push(ithPropertyObject)
+            Object.assign(
+                this.params,
+                getPropertyParams(i, ithPropertyObject)
+            )
+        }
+    }
+
+    checkParameters() {
+        return super.checkParameters()
+    }
+
+    assignParameters(): void {
+        super.assignParameters()
+
+        for (let i = 0; i < MAXIMUM_ALLOWED_PROPERTIES; i++) {
+            this.propertyObjects[i].property = this.params[`property${i}`]
+                .value as Property
+            this.propertyObjects[i].visualization = this.params[`prop${i}Vis`]
+                .value as PropertyVisualization
+            this.propertyObjects[i].color = this.params[`prop${i}Color`]
+                .value as string
+            this.propertyObjects[i].aux = this.params[`prop${i}Aux`]
+                .value as bigint
+        }
+    }
+
+    setup(): void {
+        this.setPresets()
+        this.setOverridingSettings()
+
+        console.log('Background', this.backgroundColor)
```

Comment:
This needs to be removed.

Reply:
will do in the commit I am working on. Thanks.

Reply:
Looks good!